utils/dataset.py

```diff
@@ -13,16 +13,27 @@ def __init__(self, name="train", path="fnc-1"):
         articles = self.read(bodies)
         self.articles = dict()
 
+        unrelated_stance = []
+        related_stance = []
         #make the body ID an integer value
         for s in self.stances:
             s['Body ID'] = int(s['Body ID'])
+            if s['Stance'] == 'unrelated':
+                unrelated_stance.append(s)
+            else:
+                related_stance.append(s)
+
+        self.related_stance = related_stance
+        self.unrelated_stance = unrelated_stance
 
         #copy all bodies into a dictionary
         for article in articles:
             self.articles[int(article['Body ID'])] = article['articleBody']
 
         print("Total stances: " + str(len(self.stances)))
         print("Total bodies: " + str(len(self.articles)))
+        print("Total unrelated stances: " + str(len(self.unrelated_stance)))
+        print("Total related stances: " + str(len(self.related_stance)))
```

Comment:
Instead of creating a new array and using the `append` function, it might be more concise use the filter approach: https://stackoverflow.com/questions/5640630/array-filter-in-python